UITests/HistoryTests.swift

```diff
@@ -42,21 +44,31 @@ class HistoryTests: KIFTestCase {
         EarlGrey.select(elementWithMatcher: grey_accessibilityID("url")).perform(grey_tap())
         EarlGrey.select(elementWithMatcher: grey_accessibilityLabel("History")).perform(grey_tap())
         
-        let topConstraint = GREYLayoutConstraint(attribute: GREYLayoutAttribute.top,
-                                                 relatedBy: GREYLayoutRelation.lessThanOrEqual,
-                                                 toReferenceAttribute: GREYLayoutAttribute.bottom,
-                                                 multiplier: 1.0,
-                                                 constant: 0.0)
+        // Wait until the dialog shows up
+        let listAppeared = GREYCondition(name: "Wait the history list to appear", block: { _ in
+            var errorOrNil: NSError?
+            let matcher = grey_allOf([grey_accessibilityLabel("Page 2"),
+                                      grey_sufficientlyVisible()])
+            EarlGrey.select(elementWithMatcher: matcher)
+                .inRoot(grey_accessibilityID("History List"))
+                .assert(grey_notNil(), error: &errorOrNil)
+            let success = errorOrNil == nil
+            return success
```

Comment:
nit: later in the code `return error == nil` is used, could just do that here too for consistency